sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -691,7 +692,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    new Path(getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      System.getProperty("user.dir"))).toUri.toString
```

Comment:
@avulanov Is this comment addressed ? It looks to me that its a valid concern that doing `toUri.toString` would introduce encoding that will mean we cant use `warehousePath` as `new Path(warehousePath)` anymore ?